apps/api/src/app/events/e2e/trigger-event.e2e.ts

```diff
@@ -1553,41 +1357,313 @@ describe(`Trigger event - ${eventTriggerPath} (POST)`, function () {
+                      step: firstStepUuid,
+                    },
+                  ],
+                },
+              ],
+            },
+          ],
+        });
 
-      await executionDetailsRepository.create({
-        _jobId: delayedJob!._parentId,
-        _messageId: messages[0]._id,
+        await axiosInstance.post(
+          `${session.serverUrl}${eventTriggerPath}`,
+          {
+            name: template.triggers[0].identifier,
+            to: [subscriber.subscriberId],
+            payload: {
+              customVar: 'Testing of User Name',
+            },
+          },
+          {
+            headers: {
+              authorization: `ApiKey ${session.apiKey}`,
+            },
+          }
+        );
+
+        await session.awaitRunningJobs(template?._id, true, 1);
+
+        const delayedJob = await jobRepository.findOne({
+          _environmentId: session.environment._id,
+          _templateId: template._id,
+          type: StepTypeEnum.DELAY,
+        });
+
+        expect(delayedJob!.status).to.equal(JobStatusEnum.DELAYED);
+
+        const messages = await messageRepository.find({
+          _environmentId: session.environment._id,
+          _subscriberId: subscriber._id,
+          channel: StepTypeEnum.EMAIL,
+        });
+
+        expect(messages.length).to.equal(1);
+
+        await executionDetailsRepository.create({
+          _jobId: delayedJob!._parentId,
+          _messageId: messages[0]._id,
+          _environmentId: session.environment._id,
+          _organizationId: session.organization._id,
+          webhookStatus: EmailEventStatusEnum.OPENED,
+        });
+
+        await session.awaitRunningJobs(template?._id, true, 0);
+
+        const messagesAfter = await messageRepository.find({
+          _environmentId: session.environment._id,
+          _subscriberId: subscriber._id,
+          channel: StepTypeEnum.EMAIL,
+        });
+
+        expect(messagesAfter.length).to.equal(1);
+      });
+    });
+  });
+  describe('Trigger Event - [IS_MULTI_PROVIDER_CONFIGURATION_ENABLED=true]', function () {
+    beforeEach(async () => {
+      process.env.IS_MULTI_PROVIDER_CONFIGURATION_ENABLED = 'true';
+      process.env.LAUNCH_DARKLY_SDK_KEY = '';
+      session = new UserSession();
+      await session.initialize();
+    });
+
+    afterEach(async () => {
+      process.env.IS_MULTI_PROVIDER_CONFIGURATION_ENABLED = ORIGINAL_IS_MULTI_PROVIDER_CONFIGURATION_ENABLED;
+    });
+
+    it('should trigger message with override integration identifier', async function () {
+      const newSubscriberId = SubscriberRepository.createObjectId();
+      const channelType = ChannelTypeEnum.EMAIL;
+
+      template = await createTemplate(session, channelType);
+
+      await sendTrigger(session, template, newSubscriberId);
+
+      await session.awaitRunningJobs(template._id);
+
+      const createdSubscriber = await subscriberRepository.findBySubscriberId(session.environment._id, newSubscriberId);
```

Comment:
@LetItRock There is the test, sorry, I should have highlighted it with a comment.